adafruit_ble/__init__.py

```diff
@@ -222,6 +222,10 @@ def start_scan(self, *advertisement_types, buffer_size=512, extended=False, time
             for possible_type in advertisement_types:
                 if possible_type.matches(entry) and issubclass(possible_type, adv_type):
```

Comment:
I'm confused why you still have a subclass check here if `adv_type` must match something that is exactly in `advertisement_types`.

Reply:
advertisement_types could have two related classes (including Advertisement for example) and we should return the subclass in that case.

Reply:
Ah, ok!